src/main/java/hudson/plugins/cmake/CToolBuilder.java

```diff
@@ -143,53 +146,41 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
+        }
 
-            /* Invoke tool in working dir */
-            ArgumentListBuilder cmakeCall = buildToolCall(bindir + getToolId(),
-                    Util.replaceMacro(getArguments(), envs));
-            final int exitCode;
-            if (0 != (exitCode = launcher.launch().pwd(theWorkDir).envs(envs)
-                    .stdout(listener).cmds(cmakeCall).join())) {
-                // should this failure be ignored?
-                if (ignoredExitCodes != null) {
-                    if (ignoredExitCodesParsed == null) {
-                        // parse and cache
-                        final IntSet ints = new IntSet();
-                        ints.setValues(ignoredExitCodes);
-                        ignoredExitCodesParsed = ints;
-                    }
-                    for (Iterator<Integer> iter = ignoredExitCodesParsed
-                            .iterator(); iter.hasNext();) {
-                        if (exitCode == iter.next()) {
-                            // ignore this failure exit code
-                            listener.getLogger().printf(
-                                    "%1s exited with failure code %2$s, ignored.%n",
-                                    getToolId(), exitCode);
-                            return true; // no failure
-                        }
+        /* Invoke tool in working dir */
+        ArgumentListBuilder cmakeCall = buildToolCall(bindir + getToolId(),
+            Util.replaceMacro(getArguments(), envVars));
+        final int exitCode;
+        if (0 != (exitCode = launcher.launch().pwd(theWorkDir).envs(envVars)
+            .stdout(listener).cmds(cmakeCall).join())) {
+            // should this failure be ignored?
+            if (ignoredExitCodes != null) {
+                if (ignoredExitCodesParsed == null) {
+                    // parse and cache
+                    final IntSet ints = new IntSet();
+                    ints.setValues(ignoredExitCodes);
+                    ignoredExitCodesParsed = ints;
+                }
+                for (Iterator<Integer> iter = ignoredExitCodesParsed
+                    .iterator(); iter.hasNext();) {
+                    if (exitCode == iter.next()) {
+                        // ignore this failure exit code
+                        listener.getLogger().printf(
+                            "%1s exited with failure code %2$s, ignored.%n",
+                            getToolId(), exitCode);
                     }
-                    // invocation failed, not ignored
-                    listener.getLogger().printf(
-                            "%1s exited with failure code %2$s%n", getToolId(),
-                            exitCode);
                 }
-                return false; // invocation failed
+                // invocation failed, not ignored
+                throw new AbortException(getToolId() + " exited with failure code " + exitCode);
             }
-        } catch (IOException e) {
```

Comment:
Oh please, don't swallow IOException 👎 Now the unit test is actually failing as I would expect Jenkins to fail if cmake was not available.